src/GitHub.VisualStudio/UI/Views/PullRequestCreationView.xaml.cs

```diff
@@ -23,6 +23,18 @@ public PullRequestCreationView()
                 d(ViewModel.CancelCommand.Subscribe(_ => NotifyCancel()));
                 d(ViewModel.CreatePullRequest.Subscribe(_ => NotifyDone()));
             });
+
+            IsVisibleChanged += (s, e) =>
+            {
+                if(!IsVisible)
+                    ClearForm();
+            };
+        }
+
+        private void ClearForm()
+        {
+            titleText.Text = string.Empty;
+            descriptionText.Text = string.Empty;
```

Comment:
This should all probably be done in the ViewModel: the View should only contain logic for "view" related stuff but this is all related to data. Putting stuff at the view model level also means that it can be tested with unit tests.

In `PullRequestCreationViewModel` there is `CancelCommand`: you can subscribe to this to detect when the user clicked the "Cancel" button and clear the title and description there. That way you can also create a unit test to make sure that the functionality doesn't break in future; views are impossible to unit test because you need to run the UI...